### PR TITLE
New Select: Use virtual list

### LINKED ISSUE #88898: New Select: Present a list of options in a dropdown (FAST!)
- should use [downshift](https://github.com/downshift-js/downshift) to render a dropdown of options
  - should be virtualised
  - should be 'fast'
  - Should take into account a `description` possibility

Using the [example for virtualized list](https://www.downshift-js.com/use-combobox#virtualizing-items-with-react-virtual) may be helpful

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Uses a virtual list to render items. Adds a story with those options. Can't actually use Storybook controls for it, since it crashes.

**Why do we need this feature?**

To be able to handle large data sets.

**Who is this feature for?**

All select enthusitast.

**Which issue(s) does this PR fix?**:

Fixes #88898

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tskarhed** [comment]
I tried search with 1 million items, and it was a bit laggy while typing. Maybe debouncing the input would help a bit there.

File: packages/grafana-ui/src/components/Combobox/Combobox.tsx [hunk 1]
```diff
@@ -33,32 +36,79 @@ function itemFilter(inputValue: string) {
...
-            return (
-              <li key={item.value} {...getItemProps({ item, index })}>
-                {item.label}
-              </li>
-            );
-          })}
+      <ul className={styles.ulMenu} {...getMenuProps({ ref: listRef })}>
+        {isOpen && (
+          <>
+            <li key="total-size" style={{ height: rowVirtualizer.totalSize }} />
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Combobox/Combobox.tsx:78]: Hmm.... this is a bit odd...

Should/could this instead be a height on the `<ul>` ?

Comment (joshhunt) [reply packages/grafana-ui/src/components/Combobox/Combobox.tsx:78]: Ahhhh, I think I see.

I think, any way, we would want to have a `<div>` that is the dropdown (max-height on this), and inside that a `<ul>` which has it's appropriate height.

Comment (tskarhed) [reply packages/grafana-ui/src/components/Combobox/Combobox.tsx:78]: I agree, it makes more sense.

File: packages/grafana-ui/package.json [hunk 2]
```diff
@@ -99,6 +99,7 @@
     "react-table": "7.8.0",
     "react-transition-group": "4.4.5",
     "react-use": "17.5.0",
+    "react-virtual": "^2.10.4",
```
Comment (tskarhed) [inline packages/grafana-ui/package.json:102]: Change this for `@tanstack/react-virtual`

**Review by joshhunt — APPROVED** [review]
Gave it a spin, looks good to me!

Class name is a pretty minor nit.

**Pull request merged**